lib/evss/letters/download_service.rb

```diff
@@ -0,0 +1,36 @@
+# frozen_string_literal: true
+require 'common/client/base'
+require 'common/exceptions/internal/record_not_found'
+require 'common/exceptions/external/gateway_timeout'
+
+module EVSS
+  module Letters
+    class DownloadService < EVSS::Service
+      configuration EVSS::Letters::DownloadConfiguration
+
+      def download_letter(type, options = nil)
+        with_monitoring do
+          response = if options.blank?
+                       perform(:get, type)
+                     else
+                       perform(:post, "#{type}/generate", options, 'Content-Type' => 'application/json')
```

Comment:
It's not obvious to me why a non blank `options` method param translates to adding `/generate` to the uri and an `application/json` header.  For readability, maybe wrap those performs in their own methods?